src/compiler/checker.ts

```diff
@@ -10564,6 +10565,7 @@ namespace ts {
         function getTypeFromConditionalTypeNode(node: ConditionalTypeNode): Type {
             const links = getNodeLinks(node);
             if (!links.resolvedType) {
+                links.resolvedType = resolvingConditionalType;

```

Comment:
Usually we use a combination of `pushTypeResolution` and `popTypeResolution`, this way the error case when entering and exiting the recursion is explicitly handled. It should _probably_ also report a circularity error using `reportCircularityError`, or `error(symbol.valueDeclaration, Diagnostics._0_is_referenced_directly_or_indirectly_in_its_own_type_annotation, name);` directly.

Reply:
And a repro for the test harness would be swell, since this is exactly the kind of thing that could randomly go away in another bugfix without a test case demonstrating how it can occur. (In fact, without an isolated repro, it's hard to even know if this is the right place to insert a break)

Reply:
Yeah, totally agree, but I just can’t make sense out of the repro provided :/